### DISCUSSION TITLE
Specifity with media queries

### DISCUSSION THREAD

**Discussion by edamascus** [facebook/stylex#311] [Q&A] [answered]
In the [Deterministic resolution](https://stylexjs.com/docs/learn/thinking-in-stylex/#deterministic-resolution) section, it is mentioned that in StyleX "*The last style applied always wins.*". Does this also work for declarations within media queries?

For example, if I have the following code:

```js
const md = '@media (min-width: 769px)';

const styles = stylex.create({
  base: {
    padding: {
      default: 20,
      [md]: 40,
    },
  },
  extended: {
    paddingBlockEnd: 120,
  },
});
```

When applying the styles to the element as follows:

```js
<div {...stylex.props(styles.base, styles.extended)}>
```

I expected the `paddingBlockEnd: 120` to always win because it is the last applied style. However, the `paddingBlockEnd` that wins in `[md]` is `40` and the `120` only wins in `default`. 

When I include the same media queries as follows, the `paddingBlockEnd: 120` wins in both cases:

```js
const styles = stylex.create({
  base: {
    padding: {
      default: 20,
      [md]: 40,
    },
  },
  extended: {
    paddingBlockEnd: {
      default: 120,
      [md]: 120,
    },
  },
});
```

**Comment by nmn** [comment]
> I expected the paddingBlockEnd: 120 to always win 

This is the expected bahavior.

> However, the paddingBlockEnd that wins in [md] is 40 and the 120 only wins in default.

This seems like a bug. Let me investigate.

**Reply by nmn** [reply]
QQs: 
1. What version of StyleX are you using?
2. Are you seeing the bug during development or also for production builds?

**Reply by nmn** [reply, marked as answer]
I just tried to reproduce this issue and it's working as expected. You're like on v0.3.0 and seeing this issue during development. The issue is specific to the runtimeInjection of styles in 0.3.0 AFAIK.

**Reply by edamascus** [reply]
I am on version `0.3.0` as I [had issues](https://github.com/facebook/stylex/issues/47#issuecomment-1872532723) when updating the project to the latest version.

I tested the production build and I can confirm that this is working as expected. The issue only appears in development.

**Reply by nmn** [reply]
Thanks for the confirmation.

I'm working to create a 0.5.0 beta within a few days which should fix the Vite problem.
